<launch>

    <param name="robot_name" value="sim01" />

    <node name="check_obstacle_safety_with_parameters" pkg="miro" type="check_obstacle_safety_with_parameters.py" />
    
    <node name ="new_obstacle_detection_only_flags" pkg="miro" type="new_obstacle_detection_only_flags.py" />

    <node name="odometry_publisher_from_gazebo_vel" pkg="miro" type="odometry_publisher_from_gazebo_vel.py" />

    <node name="obstacle_avoidance_with_parameters" pkg="miro" type="obstacle_avoidance_with_parameters.py" > 
      <param name="drive_pattern" value="obstacle_avoidance" />
      <param name="x_threshold" value="0.2" />
      <param name="y_threshold" value="0.05" />
      <param name="th_threshold_first_rotation" value="0.05" />
      <param name="K_first_rotation" value="1" />
      <param name="d_ref_wall_following" value="0.5" />
      <param name="K_wall_following" value="1" />
      <param name="v_wall_following" value="100" />
    </node>

 <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <node name="rqt_plot_position" pkg="rqt_plot" type="rqt_plot"
    args="/odom/pose/pose/position" />

</launch>
